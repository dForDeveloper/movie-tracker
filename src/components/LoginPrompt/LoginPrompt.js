import React from 'react';
import { Link } from 'react-router-dom';

const LoginPrompt = () => {
  return (
    <div className="login-prompt">
      <Link to='/'><button>X</button></Link>
      <h3 className="prompt-text">
        You must be logged in to save favorites.
      </h3>
      <Link to='/sign-up'><button>Sign Up</button></Link>
      <Link to='/login'><button>Log In</button></Link>
    </div>
  );
}

export default LoginPrompt;