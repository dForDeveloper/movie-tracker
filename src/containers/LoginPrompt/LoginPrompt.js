import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { toggleLoginPrompt } from '../../actions';

export const LoginPrompt = (props) => {
  const { toggleLoginPrompt } = props;
  return (
    <div className="LoginPrompt">
      <button className="close-button" onClick={() => toggleLoginPrompt(false)}>X</button>
      <div className="button-container">
        <h3 className="login-title">
          You must be logged in to save favorites.
        </h3>
        <Link to='/sign-up'>
          <button className="signupprompt-button" onClick={() => toggleLoginPrompt(false)}>Sign Up</button>
        </Link>
        <Link to='/login'>
          <button className="loginprompt-button" onClick={() => toggleLoginPrompt(false)}>Log In</button>
        </Link>
      </div>
    </div>
  );
}

export const mapDispatchToProps = (dispatch) => ({
  toggleLoginPrompt: (validity) => dispatch(toggleLoginPrompt(validity))
})

export default connect(null, mapDispatchToProps)(LoginPrompt);

LoginPrompt.propTypes = {
  toggleLoginPrompt: PropTypes.func
}