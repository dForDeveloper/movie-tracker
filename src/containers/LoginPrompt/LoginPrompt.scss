.LoginPrompt {
  z-index: 1;
  top: 30%;
  left: 35%;
  height: 170px;
  width: 430px;
  position: absolute;
  display: inline-block;
  overflow: hidden;
  text-align: center;
  background-color: rgba(243, 241, 255, 0.9);
  border: 6px solid $steel-grey;
}

.close-button,
.loginprompt-button,
.signupprompt-button {
  font-weight: bold;
  border: 2px solid $border-color;
  cursor: pointer;
}

.close-button {
  float: left;
  margin: 5px;
  padding: 2% 6%;
}

.login-title {
  font-size: 22px;
}

.button-container {
  position: relative;
  text-align: center;
  margin: 12% auto;
}

.loginprompt-button,
.signupprompt-button {
  margin: 20px;
  padding: 3% 8%;
  display: inline-block;
  font-size: 20px;
  font-weight: bold;
}

.loginprompt-button:hover,
.signupprompt-button:hover {
  background-color: $salmon;
  color: white;
}

.close-button:hover {
  background-color: $text-color;
  color: $off-white;
}

@media screen and (max-width: 1080px) {
  .LoginPrompt {
    left: 30%;
  }
}

@media screen and (max-width: 720px) {
  .LoginPrompt {
    left: 19%;
  }
}

@media screen and (max-width: 550px) {
  .LoginPrompt {
    left: 11%;
    width: 80%;
    height: 160px;
  }

  .close-button {
    float: left;
    margin: 5px;
    padding: 1% 2%;
  }
  
  .login-title {
    font-size: 16px;
    padding-top: 11px;
  }

  .loginprompt-button,
  .signupprompt-button {
    margin: 10px 15px;
    padding: 2% 8%;
  }
}