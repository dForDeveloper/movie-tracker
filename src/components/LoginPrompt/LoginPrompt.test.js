import React from 'react';
import { shallow } from 'enzyme';
import LoginPrompt from './LoginPrompt.js';

describe('LoginPrompt', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <LoginPrompt />
    );
  });

  describe('LoginPrompt component', () => {
    it('should properly render the component elements', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
