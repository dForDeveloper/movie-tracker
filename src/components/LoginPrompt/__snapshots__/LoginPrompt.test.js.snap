// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPrompt LoginPrompt component should properly render the component elements 1`] = `
<div
  className="login-prompt"
>
  <Link
    replace={false}
    to="/"
  >
    <button>
      X
    </button>
  </Link>
  <h3
    className="prompt-text"
  >
    You must be logged in to save favorites.
  </h3>
  <Link
    replace={false}
    to="/sign-up"
  >
    <button>
      Sign Up
    </button>
  </Link>
  <Link
    replace={false}
    to="/login"
  >
    <button>
      Log In
    </button>
  </Link>
</div>
`;
